naas/models/project_property.py

```diff
@@ -0,0 +1,82 @@
+import iso8601
+from naas.models.data_type import DataType
+from naas.models.data_types import DataTypes
+from naas.models.links import Links
+
+
+class ProjectProperty(object):
+    """
+
+    ProjectProperty
+    ===============
+
+    This returns an instance of the ProjectProperty domain model
+    """
+
+    def __init__(self, attributes={}):
+        self.attributes = attributes
+
+    def _id(self):
+        """Returns the data type id"""
+        return self.attributes.get('id')
+
+    def project_id(self):
+        """Returns the associated project id"""
+        return self.attributes.get('project_id')
+
+    def data_type_id(self):
+        """Returns the associated project id"""
+        return self.attributes.get('data_type_id')
+
+    def data_type_attributes(self):
+        """Returns the associated project id"""
+        return self.attributes.get('data_type', {})
+
+    def data_type(self):
+        """Returns the data type model"""
+        if self.data_type_attribute():
+            return naas.models.DataType(self.data_type_attributes)
+        return naas.models.DataTypes.retrieve(self.data_type_id())
+
+    def name(self):
+        """Returns the data type name"""
+        return self.attributes.get('name')
+
+    def key_name(self):
+        """Returns the key name"""
+        return self.attributes.get('key_name')
+
+    def description(self):
+        """Returns the data type description"""
+        return self.attributes.get('description')
+
+    def created_at(self):
+        """Returns the created at timestamp"""
+        return iso8601.parse_date(self.attributes.get('created_at'))
```

Comment:
@arthuroe - we might want to handle the case where `created_at` (or other timestamps) could be empty. While the API will have these set, we would want to ensure this model works as a standalone object.